project/Teacher.cs

```diff
@@ -48,7 +48,10 @@ public void AssignSubject(Subject subject)
                 );
 
             _subjects.Add(subject);
-            //FIX: subject.Teacher = this;
+            if (subject.Teacher != this)
```

Comment:
So we're adding or updating on the subject teacher despite us not allowing to add an existing subject in the previous lines?